independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/SubclassGenerator.java

```diff
@@ -138,9 +138,35 @@ protected FieldDescriptor createConstructor(ClassOutput classOutput, BeanInfo be
                         parameterTypes.subList(0, superParamsSize).toArray(new String[0])),
                 constructor.getThis(), superParams);
 
+        // we build a map for each interceptor instance created, so that they are shared
+        // Map<String, Object> where InjectableInterceptor.getIdentifier() is key and Object is instance of the interceptor for this bean
+        FieldCreator interceptorInstanceMap = subclass.getFieldCreator("interceptorInstanceMap", Map.class.getName())
```

Comment:
I think there is no need to add the map as a class field. It should be a local variable instead...